resources/ec2-spot-instance-request.go

```diff
@@ -0,0 +1,64 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+type EC2SpotInstanceRequest struct {
+	svc   *ec2.EC2
+	id    string
+	state string
+}
+
+func init() {
+	register("EC2SpotInstanceRequest", ListEC2SpotInstanceRequests)
+}
+
+func ListEC2SpotInstanceRequests(sess *session.Session) ([]Resource, error) {
+	svc := ec2.New(sess)
+
+	resp, err := svc.DescribeSpotInstanceRequests(nil)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This only returns one page and might not work for people who have many requests. You need to add a paginator like [here](https://github.com/rebuy-de/aws-nuke/blob/main/resources/ec2-instances.go#L45:L51) so that it actually gets all spot instance requests.